wazo_confgend/generators/extensionsconf.py

```diff
@@ -195,47 +197,44 @@ def generate(self, output):
                 if option_name == 'objtpl':
                     tmpl.append(option_value)
                     continue
-                print >> options, "%s = %s" % (option_name, option_value.replace('%%CONTEXT%%', ctx['name']))
+                ast_writer.write_option(option_name, option_value.replace('%%CONTEXT%%', context_name))
 
             # context includes
-            for row in asterisk_conf_dao.find_contextincludes_settings(ctx['name']):
-                print >> options, "include = %s" % row['include']
-            print >> options
+            for row in asterisk_conf_dao.find_contextincludes_settings(context_name):
+                ast_writer.write_option('include', row['include'])
+            ast_writer.write_newline()
 
             # objects extensions (user, group, ...)
-            for exten_row in asterisk_conf_dao.find_exten_settings(ctx['name']):
+            for exten_row in asterisk_conf_dao.find_exten_settings(context_name):
                 exten_generator = extension_generators.get(exten_row['type'], GenericExtensionGenerator)
                 exten = exten_generator(exten_row).generate()
-                self.gen_dialplan_from_template(tmpl, exten, options)
+                self.gen_dialplan_from_template(tmpl, exten, ast_writer)
 
-            self._generate_hints(ctx['name'], options)
+            self._generate_hints(ctx['name'], output)
```

Comment:
why this method is not called with `ast_writer`?

Reply:
Ah ok nvm it wasn't use print syntax so can be considered out of scope

Reply:
Yeah, there was a few places I couldn't use it because they wrote lines in formats unsupported by ast_writer or the entire line (or multiple lines) are already generated by another method. The `_write_line` is internal. I could technically still call it or rename it and make it "public" like the other methods if you'd like.

Reply:
Ah ok it's fine. Thanks for precision